src/common/reducers/actions.js

```diff
@@ -1343,6 +1345,76 @@ export const startListener = () => {
               [newInstanceName]: { ...config, name: newInstanceName }
             }
           });
+
+          const instanceManagerModalIndex = newState.modals.findIndex(
+            x =>
+              x.modalType === 'InstanceManager' &&
+              x.modalProps.instanceName === oldInstanceName
+          );
+
+          const firstPartSplittedArr = newState.modals.slice(
+            instanceManagerModalIndex
+          );
+
+          const lastPartSplittedArr2 = newState.modals.slice(
+            instanceManagerModalIndex + 1,
+            newState.modals.length
+          );
+
+          if (instanceManagerModalIndex >= 0) {
+            if (newState.modals.length === 1) {
+              dispatch({
+                type: UPDATE_MODAL,
+                modals: [
+                  {
+                    modalType: 'InstanceManager',
+                    modalProps: { instanceName: newInstanceName }
+                  }
+                ]
+              });
+            } else if (newState.modals.length === 2) {
+              if (instanceManagerModalIndex === newState.modals.length) {
+                dispatch({
+                  type: UPDATE_MODAL,
+                  modals: [
+                    ...newState.modals,
+                    {
+                      modalType: 'InstanceManager',
+                      modalProps: { instanceName: newInstanceName }
+                    }
+                  ]
+                });
+              } else {
+                dispatch({
+                  type: UPDATE_MODAL,
+                  modals: [
+                    {
+                      modalType: 'InstanceManager',
+                      modalProps: { instanceName: newInstanceName }
+                    },
+                    newState.modals[1]
+                  ]
+                });
+              }
+            } else if (instanceManagerModalIndex !== newState.modals.length) {
+              dispatch({
+                type: UPDATE_MODAL,
+                modals: [
+                  ...firstPartSplittedArr,
+                  newState.modals[instanceManagerModalIndex],
+                  ...lastPartSplittedArr2
+                ]
+              });
+            } else {
+              dispatch({
+                type: UPDATE_MODAL,
+                modals: [
+                  ...firstPartSplittedArr,
+                  newState.modals[instanceManagerModalIndex]
+                ]
+              });
+            }
+          }
```

Comment:
If we can assume there is only 1 modal per type opened at the same time, all of this can be simplified to something like
```
[...arr.slice(0, pos), newElement, ...arr.slice(pos + 1)]
```